src/app/components/scheduleform/page.tsx

```diff
@@ -0,0 +1,97 @@
+"use client";
+
+import { Button, Card } from "@tremor/react";
+import { useState } from "react";
+import { makeSchedule } from "@/app/components/scheduleform/actions";
+
+const Scheduleform = ({ property_id }: { property_id: number }) => {
+    const [input, setInput] = useState({
+        date: "",
+        start: "",
+        end: "",
+    });
+    const [error, setError] = useState<string | null>(null);
+
+    const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+        const { name, value } = e.target;
+        setInput((prev) => ({ ...prev, [name]: value }));
+    };
+
+    const handleSubmit = async (e: React.FormEvent<HTMLFormElement>) => {
+        e.preventDefault();
+        if (property_id) {
+            const startDateTime = `${input.date}T${input.start}`;
+            const endDateTime = `${input.date}T${input.end}`;
+            const output = await makeSchedule(
+                property_id,
+                new Date(startDateTime),
+                new Date(endDateTime)
+            )
+            if (typeof output === "string") {
+                setError(output);
+            } else {
+                setError(null);
+                window.location.reload();
```

Comment:
Better to use `router.refresh()` here, since it plays nicer with the slightly weird way routing works in Next.js. See the [`useRouter` documentation](https://nextjs.org/docs/app/api-reference/functions/use-router) for more info.

Reply:
sure